pkg/provider/identity.go

```diff
@@ -332,16 +357,14 @@ func (fi *FullIdentity) DialOption() (grpc.DialOption, error) {
 		InsecureSkipVerify: true,
 		VerifyPeerCertificate: peertls.VerifyPeerFunc(
```

Comment:
can we do something like this so that we have a way to get a dial option without peer ID verification?:
```
	pcvFuncs := []peertls.PeerCertVerificationFunc{peertls.VerifyPeerCertChains}
	if id != "" {
		pcvFuncs = append(pcvFuncs, VerifyPeerIdentity(id))
	}

	tlsConfig := &tls.Config{
		Certificates:       []tls.Certificate{*c},
		InsecureSkipVerify: true,
		VerifyPeerCertificate: peertls.VerifyPeerFunc(pcvFuncs...),
	}
```